compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,270 @@
+
+  while (getline(ss, str_buf, ','))
+  {
+    if (str_buf.length() && check_input(str_buf)) // input validation
+    {
+      try
+      {
+        if (str_buf.find('-') == std::string::npos) // if token has no '-'
+          by_id.push_back(stoi(str_buf));
+        else // tokenize again by '-'
+        {
+          std::istringstream ss2(str_buf);
+          std::string token;
+          int from_to[2], top = 0;
+
+          while (getline(ss2, token, '-'))
+            from_to[top++] = stoi(token);
+
+          for (int number = from_to[0]; number <= from_to[1]; number++)
+            by_id.push_back(number);
+        }
+      }
+      catch (std::invalid_argument &error)
+      {
+        std::cerr << "ERROR: [circle-opselector] Invalid argument.(stoi)" << std::endl;
+        exit(1);
+      }
+      catch (std::out_of_range)
+      {
+        std::cout << "ERROR: [circle-opselector] Argument is out of range(stoi)\n";
+        exit(1);
+      }
+      catch (...)
+      {
+        std::cout << "ERROR: [circle-opselector] Unknown error(stoi)\n";
+        exit(1);
+      }
+    }
+  }
+}
+
+void split_name_input(const std::string &str, std::vector<std::string> &by_name)
+{
+  std::istringstream ss;
+  ss.str(str);
+  std::string str_buf;
+
+  while (getline(ss, str_buf, ','))
+    by_name.push_back(str_buf);
+}
+
+int entry(int argc, char **argv)
+{
+  // TODO Add new option names!
+
+  arser::Arser arser("circle-opselector provides selecting operations in circle model");
+
+  arser.add_argument("--version")
+    .nargs(0)
+    .required(false)
+    .default_value(false)
+    .help("Show version information and exit")
+    .exit_with(print_version);
+
+  // TODO Add new options!
+
+  arser.add_argument("--input").nargs(1).type(arser::DataType::STR).help("Input circle model");
+  arser.add_argument("--output").nargs(1).type(arser::DataType::STR).help("Output circle model");
+
+  // select option
+  arser.add_argument("--by_id")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Input operation id to select nodes.");
+  arser.add_argument("--by_name")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Input operation name to select nodes.");
+  arser.add_argument("--select")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Selecte opeartors from the input circle");
+  arser.add_argument("--deselect")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Exclude operators from the input circle");
+
+  try
+  {
+    arser.parse(argc, argv);
+  }
+  catch (const std::runtime_error &err)
+  {
+    std::cout << err.what() << std::endl;
+    std::cout << arser;
+    return EXIT_FAILURE;
+  }
+
+  std::string input_path = arser.get<std::string>("--input");
+  std::string output_path = arser.get<std::string>("--output");
+
+  std::string operator_input;
+
+  std::vector<int> by_id;
+  std::vector<std::string> by_name;
+
+  std::string op;
+  std::vector<int> oplist;
+  int select_mode = -1;
+
+  if (arser["--by_id"])
```

Comment:
IIUC, `--by-id` and `--by-name` are exclusive. Related logic is required.

Reply:
https://github.com/Samsung/ONE/blob/4214bb0a1b62e5f4883b8506c416421f222610fc/compiler/tfl-inspect/driver/Driver.cpp#L51-L56